FF13Rando/Randomizers/BattleRando.cs

```diff
@@ -198,28 +205,31 @@ public override void Randomize()
                     //Only shuffle fights which actually have candidate enemies we can add to it.
                     List<string> fightsToShuffle = shuffleCandidates.Keys.Where(k => shuffleCandidates[k].Count > 0).ToList();
-                    for (int i = availableSlots; i > 0; i--)
+                    if (fightsToShuffle.Count > 0)
                     {
-                        //Select a new random enemy to add to the pool of enemies for this group.
-                        //TODO: need to take some care about enemy ranges here, probably have to do some kind of stepped distribution and select enemies in the relevant rank pools.
-                        string shuffleIdx = RandomNum.SelectRandom(fightsToShuffle);
-                        string newCandidateEnemy = RandomNum.SelectRandomWeighted(shuffleCandidates[shuffleIdx], enemy => vanillaEnemies.Contains(enemy) ? reservedCapacity : 2 * availableSlots);
-                        if (newCandidateEnemy != null)
+                        for (int i = availableSlots; i > 0; i--)
                         {
-                            enemiesToAddToSet.Add(newCandidateEnemy);
+                            //Select a new random enemy to add to the pool of enemies for this group.
+                            //TODO: need to take some care about enemy ranges here, probably have to do some kind of stepped distribution and select enemies in the relevant rank pools.
+                            string shuffleIdx = RandomNum.SelectRandom(fightsToShuffle);
+                            string newCandidateEnemy = RandomNum.SelectRandomWeighted(shuffleCandidates[shuffleIdx], enemy => vanillaEnemies.Contains(enemy) ? reservedCapacity : 2 * availableSlots);
+                            if (newCandidateEnemy != null)
+                            {
+                                enemiesToAddToSet.Add(newCandidateEnemy);
+                            }
                         }
-                    }
 
-                    foreach (string enemyToAdd in enemiesToAddToSet)
-                    {
-                        string charaspecToAdd = enemyRando.btCharaSpec[enemyToAdd].sCharaSpec_string;
-                        if (!list.Contains(charaspecToAdd))
+                        foreach (string enemyToAdd in enemiesToAddToSet)
                         {
-                            list.Add(charaspecToAdd);
+                            string charaspecToAdd = enemyRando.btCharaSpec[enemyToAdd].sCharaSpec_string;
+                            if (!list.Contains(charaspecToAdd))
+                            {
+                                list.Add(charaspecToAdd);
+                            }
                         }
+                        charaSets[charaset].SetCharaSpecs(list);
                     }
 
-                    charaSets[charaset].SetCharaSpecs(list);
                     processedCharasets.Add(charaset);
 
                     foreach (string peer in peerCharasets)
```

Comment:
Found it, it's down here
`                           peerList.Add(peerEnemy);`
should be
`                           peerList.Add(charaspecToAdd);`